datalab/datalab_session/data_operations/input_data_handler.py

```diff
@@ -27,13 +26,9 @@ def __init__(self, basename: str, source: str = None) -> None:
     """
     self.basename = basename
     self.source = source
-    self.exit_stack = ExitStack()
-    self.fits_file = self.exit_stack.enter_context(get_fits(basename, source))
+    self.fits_file = FileCache().get_fits(basename, source)
```

Comment:
Now it just gets the file from the cache so it doesn't need the special context stuff